src/module/utilities/gtpy_decorator.cpp

```diff
@@ -929,10 +929,34 @@ GtpyDecorator::findGtParent(GtObject* obj)
     return wrapGtObject(obj->findParent<GtObject*>()).release();
 }
 #if GT_VERSION >= 0x020000
+
+int
+GtpyDecorator::getPropertyContainerSize(GtObject* obj, const QString& id)
+{
```

Comment:
To avoid code duplication you could create a function that checks if the object is valid and the property struct container can be found. If everything is fine, the container is returned, otherwise the error message is printed and the function returns a nullptr.

Reply:
I added a helping function but as it is quite specific I only added it in the cpp directly at its usage.